codeworld-server/src/ErrorSanitizer.hs

```diff
@@ -0,0 +1,88 @@
+{-# LANGUAGE LambdaCase        #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+{-
+  Copyright 2017 The CodeWorld Authors. All rights reserved.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-}
+
+module ErrorSanitizer where
+
+import           Control.Monad
+import           Data.ByteString (ByteString)
+import qualified Data.ByteString as B
+import qualified Data.ByteString.Char8 as C
+import 		 Data.Char
+import 		 Data.List
+import           Data.Tuple
+import           System.Directory
+import           System.FilePath
+import           System.IO
+import           System.IO.Temp (withSystemTempDirectory)
+import           System.Process
+import           Text.Regex.TDFA hiding (Regex)
+import           Text.Regex
+
+import Util
+
+filterStages :: [(Regex, String)]
+filterStages = [
+        (mkRegex "\226\8364\162", ""),
+        (mkRegex "\226\8364\732", ""),
+        (mkRegex "\226\8364\8482", ""),
+        (mkRegex "\\’", ""),
+        (mkRegex "IO action main", "variable main"),
+        (mkRegex "module Main", "your program"),
+        (mkRegex "main:Main", "your program"),
+        (mkRegex "Couldn't match expected type Text\\s*with actual type GHC.Types.Char",
+        	"Text requires double quotes, rather than single."),
```

Comment:
No tab characters, please.  They cause compiler warnings in current GHC versions.  Same for other uses below.